src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -45,22 +43,22 @@ public OtpUserController(String apiPrefix) {
 
     @Override
     OtpUser preCreateHook(OtpUser user, Request req) {
-        String apiKey = req.headers("x-api-key");
-        // If an api key is present an API user is attempting to create an OTP user.
-        if (apiKey != null) {
+        RequestingUser requestingUser = Auth0Connection.getUserFromRequest(req);
+        String apiKeyFromHeader = req.headers("x-api-key");
+        // If third party and an api key is present user is attempting to create an OTP user.
+        if (requestingUser.apiUser != null && apiKeyFromHeader != null) {
             // Check API key and assign user to appropriate third-party application. Note: this is only relevant for
             // instances of otp-middleware running behind API Gateway.
-            ApiUser apiUser = Persistence.apiUsers.getOneFiltered(Filters.eq("apiKeys.value", apiKey));
-            if (apiUser != null) {
-                // If API user found, assign to new OTP user.
-                user.applicationId = apiUser.id;
+            if (requestingUser.apiUser.hasToken(apiKeyFromHeader)) {
+                // If third party and using their own api key, assign to new OTP user.
+                user.applicationId = requestingUser.apiUser.id;
             } else {
-                // If API user not found, report to Bugsnag for further investigation.
-                BugsnagReporter.reportErrorToBugsnag(
-                    "OTP user created with API key that is not linked to any API user",
-                    apiKey,
-                    new IllegalArgumentException("API key not linked to API user.")
-                );
+                // If API user not found, log message and halt.
```

Comment:
This is causing the UI to not be able to persist a new user. What is the recommendation for creating an ApiUser for the UI?

Reply:
Also, it is possible, with the code as written, to POST a new OtpUser with the Bearer token but without providing an API key. Should that be allowed? (The application id for that user will be undefined.)

Reply:
@binh-dam-ibigroup I'm assuming you mean "What is the recommendation for creating an **OtpUser** for the UI?"? I think the fix to cover you second comment will allow an OtpUser to be created.

Reply:
@br648, right, It looks like no action is needed to make the UI work after your last commit.